include/nuttx/1wire/1wire_master.h

```diff
@@ -0,0 +1,179 @@
+ *            device
+ *   buflen - The number of bytes to send from the buffer
+ *
+ * Returned Value:
+ *   0: success, <0: A negated errno
+ *
+ ****************************************************************************/
+
+int onewire_write(FAR struct onewire_master_s *master,
+                  FAR const struct onewire_config_s *config,
+                  FAR const uint8_t *buffer, int buflen);
+
+/****************************************************************************
+ * Name: onewire_read
+ *
+ * Description:
+ *   Receive a block of data from 1WIRE. Each read operation will be an
+ *   'atomic' operation in the sense that any other 1WIRE actions will be
+ *   serialized and pend until this read completes.
+ *
+ * Input Parameters:
+ *   master - Device-specific state data
+ *   config - Described the 1WIRE configuration
+ *   buffer - A pointer to a buffer of data to receive the data from the
+ *            device
+ *   buflen - The requested number of bytes to be read
+ *
+ * Returned Value:
+ *   0: success, <0: A negated errno
+ *
+ ****************************************************************************/
+
+int onewire_read(FAR struct onewire_master_s *master,
+                 FAR const struct onewire_config_s *config,
+                 FAR uint8_t *buffer, int buflen);
+
+/****************************************************************************
+ * Name: onewire_writeread
+ *
+ * Description:
+ *   Receive a block of data from 1WIRE. Each read operation will be an
+ *   'atomic' operation in the sense that any other 1WIRE actions will be
+ *   serialized and pend until this read completes.
+ *
+ * Input Parameters:
+ *   master  - Device-specific state data
+ *   config  - Described the 1WIRE configuration
+ *   wbuffer - A pointer to the read-only buffer of data to be written to
+ *             device
+ *   wbuflen - The number of bytes to send from the buffer
+ *   rbuffer - A pointer to a buffer of data to receive the data from the
+ *             device
+ *   rbuflen - The requested number of bytes to be read
+ *
+ * Returned Value:
+ *   0: success, <0: A negated errno
+ *
+ ****************************************************************************/
+
+int onewire_writeread(FAR struct onewire_master_s *master,
+                      FAR const struct onewire_config_s *config,
+                      FAR const uint8_t *wbuffer, int wbuflen,
+                      FAR uint8_t *rbuffer, int rbuflen);
+
+/****************************************************************************
+ * Name: onewire_initialize
+ *
+ * Description:
+ *   Return 1-wire bus master from 1-wire lower half device.
+ *
+ * Input Parameters:
+ *   dev       - Pointer to the allocated 1-wire lower half
+ *   maxslaves - Maximum number of 1-wire slave devices
+ *
+ ****************************************************************************/
+
+FAR struct onewire_master_s *
+  onewire_initialize(FAR struct onewire_dev_s *dev, int maxslaves);
```

Comment:
remove the space at the beginning

Reply:
That's new for me and also not caught by checkpatch, but sure I can change it.